### PR TITLE
API to restart deployment

### PR DESCRIPTION
**Pull request by yunfeng-scale**
# Pull Request Summary

as title. will return error for endpoints with more than 1 node

## Test Plan and Usage Guide

unit test and tested with prod deployments

### CODE REVIEW

File: model-engine/model_engine_server/infra/services/live_model_endpoint_service.py [hunk 1]
```diff
@@ -403,5 +403,26 @@ async def delete_model_endpoint(self, model_endpoint_id: str) -> None:
...
         logger.info(f"Endpoint delete released lock for {created_by}, {name}")
 
+    async def restart_model_endpoint(self, model_endpoint_id: str) -> None:
+        record = await self.model_endpoint_record_repository.get_model_endpoint_record(
+            model_endpoint_id=model_endpoint_id
+        )
+        if record is None:
+            raise ObjectNotFoundException
+
+        async with self.model_endpoint_record_repository.get_lock_context(record) as lock:
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/infra/services/live_model_endpoint_service.py:413]: not super sure if we need the lock? the lock's been troublesome before

Comment (seanshi-scale) [reply model-engine/model_engine_server/infra/services/live_model_endpoint_service.py:413]: and if we can avoid it then it'd be nice

File: model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py [hunk 2]
```diff
@@ -1489,6 +1494,33 @@ async def _delete_keda_scaled_object(endpoint_id: str) -> bool:
...
+                    "metadata": {
+                        "annotations": {
+                            "model-engine.scale.com/restartedAt": datetime.datetime.now().isoformat()
+                        }
+                    }
+                }
+            }
+        }
+
+        try:
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py:1513]: should we be more explicit in the LeaderWorkerSet case? e.g. return that "it doesn't work for LWSs"

Comment (yunfeng-scale) [reply model-engine/model_engine_server/infra/gateways/resources/k8s_endpoint_resource_delegate.py:1513]: make sense. let me do this in the use case code

File: model-engine/model_engine_server/api/model_endpoints_v1.py [hunk 3]
```diff
@@ -207,3 +209,27 @@ async def delete_model_endpoint(
...
+    """
+    Restarts the Model endpoint.
+    """
+    logger.info(f"POST /model-endpoints/{model_endpoint_id}/restart for {auth}")
+    try:
+        use_case = RestartModelEndpointV1UseCase(
+            model_endpoint_service=external_interfaces.model_endpoint_service,
+        )
+        return await use_case.execute(user=auth, model_endpoint_id=model_endpoint_id)
+    except (ObjectNotFoundException, ObjectNotAuthorizedException) as exc:
```
Comment (seanshi-scale) [inline model-engine/model_engine_server/api/model_endpoints_v1.py:231]: we probably want to catch the ObjectHasInvalidValueException here and return 400 (or something)?

**Review by seanshi-scale — APPROVED** [review]
had a comment about catching the InvalidValueException in the API and forwarding the response code (I don't think we have anything else that catches it?), other than that LGTM

**Pull request merged**